### PR TITLE
feat: add settings page, and show placeholder if no credit

### LINKED ISSUE #8: Show placeholder image if no credit
Depending on what CMS publishers are coming from, their historical image credits may not be migrated. In those cases, displaying images without proper attribution can expose the publisher to legal liabilities.

It would be helpful if a placeholder image in the media library could be defined. If no credit is present for an image, the placeholder would then be displayed instead. This approach would allow the publisher to manually input credits as needed and have the copyrighted images reappear, without needing to remove the images from the media library or post body in the meantime.

### PR DESCRIPTION
**Pull request by dkoo**
Adds a new settings page for Image Credits options under the **WP Admin > Settings > Media** menu (placed there because this plugin has no dependencies on other Newspack products). New options available:

* **Image Credit Class Name:** Lets you set the CSS class name given to the `span` that wraps image credit strings. This defaults to `image-credit`, which was previously hard-coded, so it shouldn't change behavior for sites that are already using the plugin.
* **Image Credit Label:** Lets you change the label printed before image credits. This defaults to `Credit: `, which was previously hard-coded, so it shouldn't change behavior for sites that are already using the plugin. 
* **Placeholder Image:** Lets you choose an image from the media library to be displayed in place of any image that lacks a credit. This should help publishers avoid legal trouble in the event of messy migrations that fail to bring over all credit data. Will work for featured images, core/image blocks, and jetpack/slideshow blocks.

Screen Shot 2021-08-23 at 2 05 44 PM

_Note: The options have been moved to the Media settings page instead of living in their own standalone page._

To test:

1. The first two options (class name and label) are straightforward: test with the default options, then with updated options, and then with empty options. Verify that the settings are respected on the front-end for all three contexts (featured image, core image block, Jetpack slideshow block).
2. For the placeholder image, first verify that without choosing any image, behavior is unchanged. Images are rendered as usual with or without a credit.
3. Choose a placeholder image from the Media Library, then save.
4. Now verify that any images that lack a credit are replaced by the chosen placeholder image (just the image, not the caption or anything else having to do with the image) in all contexts.
5. Add a credit to images that are missing them and verify that after a refresh, the images render instead of the placeholder.
6. Remove the placeholder in settings, save, and verify that all images behave as usual.

### CODE REVIEW

**Comment by dkoo** [comment]
Also adds a `newspack_image_credits_media_credit` filter on the credits data output, as suggested by @benlk.

File: includes/class-newspack-image-credits-settings.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+				'description' => __( 'A CSS class name to be applied to all image credit elements. Leave blank to display no class name.', 'newspack-image-credits' ),
+				'key'         => 'newspack_image_credits_class_name',
+				'label'       => __( 'Image Credit Class Name', 'newspack-image-credits' ),
+				'type'        => 'input',
+				'value'       => __( 'image-credit', 'newspack-image-credits' ),
+			],
+			[
+				'description' => __( 'A label to prefix all image credits. Leave blank to display no prefix.', 'newspack-image-credits' ),
+				'key'         => 'newspack_image_credits_prefix_label',
+				'label'       => __( 'Image Credit Label', 'newpack-listings' ),
```
Comment (miguelpeixe) [inline includes/class-newspack-image-credits-settings.php]: Need to change the domain from `newspack-listing` to `newspack-image-credits`

File: includes/class-newspack-image-credits-settings.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+				'type'        => 'input',
+				'value'       => __( 'Credit:', 'newspack-image-credits' ),
+			],
+			[
+				'description' => __( 'A placeholder image to be displayed in place of images without credits. If none is chosen, the image will be displayed normally whether or not it has a credit.', 'newspack-image-credits' ),
+				'key'         => 'newspack_image_credits_placeholder',
+				'label'       => __( 'Placeholder Image', 'newpack-listings' ),
```
Comment (miguelpeixe) [inline includes/class-newspack-image-credits-settings.php]: Need to change the domain from `newspack-listing` to `newspack-image-credits`

File: includes/class-newspack-image-credits.php [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+	/**
+	 * Add media credit fields to the media editor.
+	 *
+	 * @param array $fields Array of media editor field info.
+	 * @param WP_Post $post Post object for current attachment.
+	 * @return array Modified $fields.
+	 */
+	public static function add_media_credit( $fields, $post ) {
+		$credit_info = self::get_media_credit( $post->ID );
+        $fields['media_credit'] = [
```
Comment (miguelpeixe) [inline includes/class-newspack-image-credits.php]: Update indentation from space to tabs.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		"eslint-config-prettier": "^6.10.1",
+		"eslint-plugin-import": "^2.20.2",
+		"eslint-plugin-jsdoc": "^22.1.0",
+		"eslint-plugin-prettier": "^3.1.2",
+		"lint-staged": "^10.1.2",
+		"npm-run-all": "^4.1.5",
+		"prettier": "https://github.com/Automattic/wp-prettier/releases/download/wp-1.16.4/wp-prettier-1.16.4.tgz",
+		"stylelint": "^13.12.0",
+		"stylelint-config-prettier": "^8.0.1",
+		"stylelint-config-wordpress": "^17.0.0",
```
Comment (miguelpeixe) [inline package.json]: Non-blocking:  `stylelint-config-wordpress` [has been deprecated](https://github.com/WordPress/gutenberg/pull/22777) in favor of [@wordpress/stylelint-config](https://www.npmjs.com/package/@wordpress/stylelint-config). Should we start using the maintained package?

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
It's working great! Crazy to see all my images being replaced for a placeholder 😅 

I've left some comments for small adjustments and there's also a tiny but very annoying UX detail: when I write the credit name to the image, I then press `tab` to start writing the next field. This action fires a save action and after it succeeds the field I'm writing loses focus and I start triggering keyboard shortcuts 😆 . I don't know why this happens only to the image credits fields.

File: includes/class-newspack-image-credits-settings.php (hunk 1 above)
Comment (dkoo) [reply includes/class-newspack-image-credits-settings.php]: Whoops! Updated in fcc3699

File: includes/class-newspack-image-credits-settings.php (hunk 2 above)
Comment (dkoo) [reply includes/class-newspack-image-credits-settings.php]: Whoops! Updated in fcc3699

File: includes/class-newspack-image-credits.php (hunk 3 above)
Comment (dkoo) [reply includes/class-newspack-image-credits.php]: Updated in fcc3699

File: package.json (hunk 4 above)
Comment (dkoo) [reply package.json]: Updated in 1fcfcbe

**Comment by dkoo** [comment]
> when I write the credit name to the image, I then press tab to start writing the next field. This action fires a save action and after it succeeds the field I'm writing loses focus and I start triggering keyboard shortcuts 😆 . I don't know why this happens only to the image credits fields.

@miguelpeixe I'm also not sure why this is, but the fields are added using the [`attachment_fields_to_edit`](https://developer.wordpress.org/reference/hooks/attachment_fields_to_edit/) and [`attachment_fields_to_save`](https://developer.wordpress.org/reference/hooks/attachment_fields_to_save/) filter hooks. I don't see any options available here to change focus/blur behavior.

**Review by miguelpeixe — APPROVED** [review]
It's probably a core issue. Thank you for the updates! 💯

**Pull request merged**